docs/extend/elasticsearch-service.md

```diff
@@ -32,3 +32,30 @@ export class MyPlugin implements Plugin {
 
 For advanced use-cases, such as a search for specific objects, use the [Global search plugin](https://github.com/elastic/kibana/blob/master/x-pack/platform/plugins/shared/global_search/README.md).
 
+### Logging
+
+The Elasticsearch service logs all requests and responses at the `debug` log level. 
+
+By default, it logs them using the logger `elasticsearch.query.data`, where `data` is the default client (other custom clients created via `core.elasticsearch.create(type, options)` are logged as `elasticsearch.query.<type>`). 
+
+However, if you want to group some queries for easier troubleshooting, you can indicate a different logger name for each request by passing the `loggingOptions` context parameter to the second parameter of each request method.
+
+```typescript
+client.search(searchParams, {
+  context: {
+    loggingOptions: {
+      loggerName: 'myqueries', // This request will be logged using the logger `elasticsearch.query.myqueries`
```

Comment:
As I'm reading the docs, should we change the feature to make it log as `elasticsearch.query.data.myqueries` instead?

I mean `elasticsearch.query.<client type>.<logger name>`.

The good: it doesn't break the current `elasticsearch.query.data` debug logger (as it will include them)
The bad: the same reason, as it makes it harder to troubleshoot what's not separately categorized.